Framework/Algorithms/src/IntegrateByComponent.cpp

```diff
@@ -186,18 +187,23 @@ IntegrateByComponent::makeMap(API::MatrixWorkspace_sptr countsWS, int parents) {
     return makeInstrumentMap(countsWS);
   }
 
+  const auto &detectorInfo = countsWS->detectorInfo();
```

Comment:
- Instead of the `try` and `catch`, use
  ```cpp
  if(!spectrumInfo.hasDetectors(i)) {
    g_log.debug("Spectrum has no detector, skipping);
    continue;
  }
  ```
- Similar to above, use `detIndex = spectrumInfo.spectrumDefinition(i)[0].first`.
- Then use `detectorInfo.detector(detIndex)`
- Variable `instrument` is unused now.